pkg/istio-agent/agent.go

```diff
@@ -406,13 +406,30 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 		return nil, fmt.Errorf("failed to start local DNS server: %v", err)
 	}
 
-	a.secretCache, err = a.newSecretManager()
-	if err != nil {
-		return nil, fmt.Errorf("failed to start workload secret manager %v", err)
-	}
+	if socketFileExists(security.WorkloadIdentitySocketPath) {
+		log.Info("SDS socket found. Istio SDS Server won't be started")
+	} else {
+		if security.CheckWorkloadCertificate(security.WorkloadIdentityCertChainPath, security.WorkloadIdentityKeyPath, security.WorkloadIdentityRootCertPath) {
+			log.Info("workload certificate files detected, creating secret manager without caClient")
```

Comment:
Do we need to check if the cert file is still valid, not expired yet?

Reply:
We didn't check the cert expiration to keep consistency with the GkeWorkloadCertificate logic, where the certificates presence is detected and they are not checked for validity.